src/rocdecode/roc_decoder_caps.h

```diff
@@ -26,83 +26,232 @@ THE SOFTWARE.
-                auto it_bitdepth_minus8 = std::find(codec_spec.bitdepth_minus8.begin(), codec_spec.bitdepth_minus8.end(), pdc->bit_depth_minus_8);
-                if (it_chroma_format != codec_spec.chroma_format.end() && it_bitdepth_minus8 != codec_spec.bitdepth_minus8.end()) {
-                    pdc->is_supported = 1;
-                    pdc->num_decoders = vcn_spec.num_decoders;
-                    pdc->output_format_mask = codec_spec.output_format_mask;
-                    pdc->max_width = codec_spec.max_width;
-                    pdc->max_height = codec_spec.max_height;
-                    pdc->min_width = codec_spec.min_width;
-                    pdc->min_height = codec_spec.min_height;
-                    return ROCDEC_SUCCESS;
-                } else {
-                    return ROCDEC_NOT_SUPPORTED;
+    rocDecStatus ProbeHwDecodeCapabilities() {
+        std::string drm_node = "/dev/dri/renderD128"; // look at device_id 0
+        int drm_fd = open(drm_node.c_str(), O_RDWR);
+        if (drm_fd < 0) {
+            ERR("Failed to open drm node." + drm_node);
+            return ROCDEC_DEVICE_INVALID;
+        }
+        amdgpu_device_handle dev_handle;
+        uint32_t major_version = 0, minor_version = 0;
+        if (amdgpu_device_initialize(drm_fd, &major_version, &minor_version, &dev_handle)) {
+		    ERR("GPU device initialization failed: " + drm_node);
+            return ROCDEC_DEVICE_INVALID;
+        }
+        if (amdgpu_query_hw_ip_count(dev_handle, AMDGPU_HW_IP_VCN_ENC, &num_codec_engines_)) {
+            ERR("Failed to get the number of video codec engines.");
+        }
+
+        VADisplay va_display = vaGetDisplayDRM(drm_fd);
+        if (!va_display) {
+            ERR("Failed to create va_display.");
+            return ROCDEC_DEVICE_INVALID;
+        }
+        vaSetInfoCallback(va_display, NULL, NULL);
+        int va_major_version = 0, va_minor_version = 0;
+        CHECK_VAAPI(vaInitialize(va_display, &va_major_version, &va_minor_version));
+
+        int num_profiles = 0;
+        std::vector<VAProfile> profile_list;
+        num_profiles = vaMaxNumProfiles(va_display);
+        profile_list.resize(num_profiles);
+        CHECK_VAAPI(vaQueryConfigProfiles(va_display, profile_list.data(), &num_profiles));
+
+        // To simplify, merge all profile attributes into one codec type.
+        rocDecVideoCodec codec_type;
+        rocDecVideoChromaFormat chroma_format;
+        int bit_depth;
+        for (int i = 0; i < num_profiles; i++) {
+            bool interested = false;
+            bit_depth = 8;
+            switch (profile_list[i]) {
+                case VAProfileH264Main:
+                case VAProfileH264High:
+                case VAProfileH264ConstrainedBaseline:
+                    codec_type = rocDecVideoCodec_AVC;
+                    chroma_format = rocDecVideoChromaFormat_420;
```

Comment:
There is also query for supported chroma format and it can return multiple chroma formats:

```c
VAConfigAttrib attrib = {
    .type = VAConfigAttribRTFormat;
};
vaGetConfigAttributes(va_display, profile_list[i],  VAEntrypointVLD, &attrib, 1);
if (attrib.value & VA_RT_FORMAT_YUV420) // rocDecVideoChromaFormat_420
if (attrib.value & VA_RT_FORMAT_YUV400) // rocDecVideoChromaFormat_Monochrome
// ...
```

Reply:
Also for bit depth: `VA_RT_FORMAT_YUV420` 8bit, `VA_RT_FORMAT_YUV420_10` 10bit.

Reply:
Right we better get the chroma format and bit depth explicitly.